### PR TITLE
FLO-1: Uncollected Protocol Fees Are Permanently Lost when Reserves Are Low

### PR DESCRIPTION
**Pull request by UlyanaAndrukhiv**
Closes: #210

## Context
This PR fixes a protocol accounting issue where insurance and stability fees could be permanently lost when reserves were insufficient during fee collection.

Our solution is to **skip fee collection until there are sufficient reserves to cover the full amount**, preventing partial withdrawals and preserving the correct accounting.

## Changes:
-  Prevented partial collection of `insurance` and `stability fees` when `reserve` does not contain enough tokens to cover the full calculated amount
- Skipped collection and preserve the accrued fees until reserves become sufficient.
- Avoided resetting `lastInsuranceCollectionTime` and `lastStabilityFeeCollectionTime` when collection is skipped
- Updated tests to match the new collection behavior
- Added small doc comments describing the edge case when reserves are insufficient according to https://github.com/onflow/FlowALP/pull/254#pullrequestreview-3930816660

### CODE REVIEW

File: cadence/contracts/FlowALPv0.cdc [hunk 1]
```diff
@@ -1935,31 +1942,35 @@ access(all) contract FlowALPv0 {
...
-            if reserveVault.balance == 0.0 {
-                tokenState.setLastInsuranceCollectionTime(currentTime)
+            if insuranceAmountUFix64 > reserveVault.balance {
+                // do not collect the insurance fee if the reserve doesn't have enough tokens to cover the full amount 
                 return nil
-            }
-
-            let amountToCollect = insuranceAmountUFix64 > reserveVault.balance ? reserveVault.balance : insuranceAmountUFix64
-            var insuranceVault <- reserveVault.withdraw(amount: amountToCollect)
+            } else {
```
Comment (jordanschalm) [inline cadence/contracts/FlowALPv0.cdc]: nit: since the if branch is very short and always returns, can we remove the else branch, so the happy path is outside the conditional. 

@claude Can you do this? Remove the `else {` on line 1948 (and the closing }, then unindent the former body of the else block.

File: cadence/contracts/FlowALPv0.cdc [hunk 2]
```diff
@@ -1986,17 +1997,14 @@ access(all) contract FlowALPv0 {
...
-            if reserveVault.balance == 0.0 {
-                tokenState.setLastStabilityFeeCollectionTime(currentTime)
+            if stabilityAmountUFix64 > reserveVault.balance {
+                // do not collect the stability fee if the reserve doesn't have enough tokens to cover the full amount 
                 return nil
+            } else {    
+                let stabilityVault <- reserveVault.withdraw(amount: stabilityAmountUFix64)
+                tokenState.setLastStabilityFeeCollectionTime(currentTime)
+                return <-stabilityVault
             }
```
Comment (jordanschalm) [inline cadence/contracts/FlowALPv0.cdc]:
```suggestion
            }

            let stabilityVault <- reserveVault.withdraw(amount: stabilityAmountUFix64)
            tokenState.setLastStabilityFeeCollectionTime(currentTime)
            return <-stabilityVault
```
